### PR TITLE
Added CREATE and DELETE playlist APIs

### LINKED ISSUE #3: Add endpoint POST /self/playlist
[See](https://github.com/defvs/connect-v2-docs/wiki/Playlists#create-a-playlist-post-selfplaylist)

- should include unit tests for validating the request in memory
- should include integration/live test for executing the request against the actual monstercat api

### LINKED ISSUE #8: Add endpoint DELETE /playlist/[playlistId]
[See](https://github.com/defvs/connect-v2-docs/wiki/Playlists#deleting-a-playlist-delete-playlistplaylistid)

- should include unit tests for validating the request in memory
- should include integration/live test for executing the request against the actual monstercat api

### PR DESCRIPTION
**Pull request by karimelazzouni**
#8 + added a CREATE API + added LiveApi UTs

### CODE REVIEW

**Comment by karimelazzouni** [comment]
This also targets #3 , missing the in memory tests, will add them soon

**Comment by karimelazzouni** [comment]
@Insire I would like this to count towards my Hacktober fest PRs, didn’t know that it had to be merged between 1-31 October. If possible, let’s not merge this until October 1st 😊

**Comment by Insire** [comment]
Sure thing

**Comment by karimelazzouni** [comment]
Force pushed change to the author email, forgot to switch away from my work email :)

File: MonstercatNet/Models/Playlist/Playlist.cs
Comment (Insire) [inline MonstercatNet/Models/Playlist/Playlist.cs]: These usings are unused. Can we remove them?

File: MonstercatNet/Models/Playlist/Playlist.cs
Comment (Insire) [inline MonstercatNet/Models/Playlist/Playlist.cs]: I'd like to cut down the namespace to the root namespace, so that users of this lib have to only import a single namespace to get all the available features.

File: MonstercatNet/Models/Playlist/PlaylistTrack.cs
Comment (Insire) [inline MonstercatNet/Models/Playlist/PlaylistTrack.cs]: I'd like to cut down the namespace to the root namespace, so that users of this lib have to only import a single namespace to get all the available features.

File: MonstercatNet/Models/Playlist/Playlist.cs
Comment (Insire) [inline MonstercatNet/Models/Playlist/Playlist.cs]: [nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references) are enabled for this repo, so this property generates a warning. Can we set an initial value here, so that it is not null?

File: MonstercatNet/Models/Playlist/Playlist.cs
Comment (Insire) [inline MonstercatNet/Models/Playlist/Playlist.cs]: [nullable reference types](https://docs.microsoft.com/en-us/dotnet/csharp/nullable-references) are enabled for this repo, so this property generates a warning. Can we set an initial value here, so that it is not null?

File: MonstercatNet/MonstercatApi.cs
Comment (Insire) [inline MonstercatNet/MonstercatApi.cs]: looks like there is a null check missing here for the request instance itself

File: MonstercatNet.Tests/LiveApiTests.cs
Comment (Insire) [inline MonstercatNet.Tests/LiveApiTests.cs]: This test fails for me, as the playlist doesnt exist.

File: MonstercatNet/MonstercatApi.cs
Comment (Insire) [inline MonstercatNet/MonstercatApi.cs]: This endpoint returns the created playlist. In order to be able to delete the playlist, we need to return that. Can you take a look at this again?

**Review by Insire — CHANGES_REQUESTED** [review]
Hi, can you take a look at this again. There are some issues that need to be adressed before i can merge this.

File: MonstercatNet/Models/Playlist/Playlist.cs
Comment (karimelazzouni) [reply MonstercatNet/Models/Playlist/Playlist.cs]: @Insire Which root namespace? I noticed that all the other models are under directories inside the Models directory.
E.g. Release, Self, Track.

Where would you like me to move the `Playlist` class?

File: MonstercatNet.Tests/LiveApiTests.cs
Comment (karimelazzouni) [reply MonstercatNet.Tests/LiveApiTests.cs]: I don't think there is any other way to test the API call in the `LiveApiTests` other than placing the playlist ID manually there first. 

Calling the `CreatePlaylist` API call first doesn't work either because Monstercat's delete API requires the playlist ID which is not returned from the create call.

One possible workaround would be to call the get playlists Monstercat API before testing this delete method, but that's out of the scope of this PR.

File: MonstercatNet/MonstercatApi.cs
Comment (karimelazzouni) [reply MonstercatNet/MonstercatApi.cs]: As explained in the comment above.

File: MonstercatNet.Tests/LiveApiTests.cs
Comment (karimelazzouni) [reply MonstercatNet.Tests/LiveApiTests.cs]: In order for this test to work for you:
1. Create a playlist in your account (either through the API or manually through the web portal)
1. While signed in, call the [get playlists API call directly to Monstercat](https://connect.monstercat.com/v2/self/playlists)
1. Copy any of your playlist's id
1. Replace the GUID in this test with the playlist's id and run the test

File: MonstercatNet/Models/Playlist/Playlist.cs
Comment (Insire) [reply MonstercatNet/Models/Playlist/Playlist.cs]: no no, dont move the class. its fine where it is, just shorten the namespace from ``namespace SoftThorn.MonstercatNet.Models.Playlist`` to ``namespace SoftThorn.MonstercatNet``

File: MonstercatNet.Tests/LiveApiTests.cs
Comment (Insire) [reply MonstercatNet.Tests/LiveApiTests.cs]: > which is not returned from the create call.

actually, you do get the information from the api. The unofficial docs are just not up to date regarding this. I have verified this with postman and you get this json as response to a successful POST

grafik

And as you can probably see, there is an ID provided there. I also verified that you can use it to delete the playlist.

Comment (karimelazzouni) [reply MonstercatNet.Tests/LiveApiTests.cs]: Great thing you verified it! Ok, will update it 👍

Comment (defvs) [reply MonstercatNet.Tests/LiveApiTests.cs]: @Insire what's that with this API field ? You can edit the docs or tell me and I'll do it.

**Review by Insire — COMMENTED** [review]
LGTM

**Review by Insire — APPROVED** [review]

**Comment by Insire** [comment]
Thanks for your contribution :)

**Pull request merged**
